File: api/auth-service/src/main/java/com/carrental/authservice/service/RegistrationService.java
package com.carrental.authservice.service;

import com.carrental.authservice.exceptions.UserAlreadyExistException;
import com.carrental.authservice.model.dto.CreateUserDTO;
import com.carrental.authservice.model.dto.UserResponseDTO;

public interface RegistrationService {

    UserResponseDTO registerUser(CreateUserDTO createUserDTO) throws UserAlreadyExistException;

    UserResponseDTO enableRegisteredUser(Long userId);
}

File: frontend-app/shared/service/RegistrationService.ts
import { CreateUserDTO, UserResponseDTO, ResponseData } from '../model';
import { fetchPut } from './FetchUtil';
import { USER_SERVICE_ENDPOINTS } from '../constant';

export class RegistrationService {
    static registerUser = (createUserDTO: CreateUserDTO): Promise<ResponseData<UserResponseDTO | CreateUserDTO>> => {
        return fetchPut<UserResponseDTO | CreateUserDTO>(USER_SERVICE_ENDPOINTS.USER_REGISTRATION, createUserDTO);
    };
}
